src/main/java/org/opentripplanner/updater/stoptime/TimetableSnapshotSource.java

```diff
@@ -860,6 +864,9 @@ private boolean validateAndHandleModifiedTrip(TimetableSnapshotSourceMetrics met
         // Check whether trip id already exists in graph
         String tripId = tripDescriptor.getTripId();
         Trip trip = getTripForTripId(feedId, tripId);
+
```

Comment:
trip can be null if there is bad data. You can either move this to after the check below or have setTripHeadsignFromUpdate deal with nulls gracefully.